### PR TITLE
Add doctest for StepStateScheduler

### LINKED ISSUE #2266: [docs] Add doctests for parameter schedulers
### 📚 Documentation

Following #2230 , I listed the parameter schedulers places (suggested by @sdesrozis) where doctests can be added.

- [x] StateParamScheduler (Abstract class, no test needed)
- [x] ConcatScheduler (#2384) 
- [x] CosineAnnealingScheduler (#2384)
- [x] CyclicalScheduler (Abstract class, no test needed)
- [x] LRScheduler (#2389)
- [x] LinearCyclicalScheduler (#2327)
- [x] ParamGroupScheduler (#2389)
- [x] ParamScheduler (Abstract class, no test needed)
- [x] PiecewiseLinear (#2290)
- [x] create_lr_scheduler_with_warmup (#2389)
- [x] LambdaStateScheduler (#2402)
- [x] PiecewiseLinearStateScheduler (#2400)
- [x] ExpStateScheduler (#2391)
- [x] StepStateScheduler (#2394)
- [x] MultiStepStateScheduler (#2399)

This is a great issue for first comers and those who want to learn more about Sphinx. This issue is the community effort.
If you want to contribute, please comment on this issue with the metric you want to add doctest (one by one).

Example on how to add doctest, see #2241 

Thanks!

### PR DESCRIPTION
**Pull request by bibhabasumohapatra**
Addresses #2266
Description:
added doctest for StepStateScheduler
Check list:

- [x] New tests are added (if a new feature is added)
- [x] New doc strings: description and/or example code are in RST format

### CODE REVIEW

File: ignite/handlers/state_param_scheduler.py [hunk 1]
```diff
@@ -329,21 +329,43 @@ class StepStateScheduler(StateParamScheduler):
...
-        .. code-block:: python
+        .. testsetup::
 
-            ...
-            engine = Engine(train_step)
+            engine = get_default_trainer()
+
+        .. testcode::
+
+            from ignite.handlers.state_param_scheduler import StepStateScheduler
```
Comment (sdesrozis) [inline ignite/handlers/state_param_scheduler.py:338]:
```suggestion
```
Useless.

In the conf.py of the doc : 
https://github.com/pytorch/ignite/blob/73b9d1ff390d4c253cc9fb440b484f9a193c2c11/docs/source/conf.py#L341

In ignite.handler.__init__.py
https://github.com/pytorch/ignite/blob/73b9d1ff390d4c253cc9fb440b484f9a193c2c11/ignite/handlers/__init__.py#L21

File: ignite/handlers/state_param_scheduler.py [hunk 2]
```diff
@@ -329,21 +329,43 @@ class StepStateScheduler(StateParamScheduler):
...
 
             param_scheduler = StepStateScheduler(
-                param_name="param", initial_value=10, gamma=0.99, step_size=5
-            )
+                    param_name="param", initial_value=1, gamma=0.9, step_size=5, create_new=True
```
Comment (sdesrozis) [inline ignite/handlers/state_param_scheduler.py:341]:
```suggestion
                param_name="param", initial_value=1, gamma=0.9, step_size=5, create_new=True
```

File: ignite/handlers/state_param_scheduler.py [hunk 3]
```diff
@@ -329,21 +329,43 @@ class StepStateScheduler(StateParamScheduler):
...
+                    )
```
Comment (sdesrozis) [inline ignite/handlers/state_param_scheduler.py:342]:
```suggestion
                )
```

File: ignite/handlers/state_param_scheduler.py [hunk 4]
```diff
@@ -329,21 +329,43 @@ class StepStateScheduler(StateParamScheduler):
...
+            # Epoch 15, param changes from 0.81 to 0.81*0.9, param = 0.729
+            # and so on ... the param change at Epoch = 5, 10, 15, 20, . . .
 
             param_scheduler.attach(engine, Events.EPOCH_COMPLETED)
 
-            # basic handler to print scheduled state parameter
-            engine.add_event_handler(Events.EPOCH_COMPLETED, lambda _ : print(engine.state.param))
+            @engine.on(Events.EPOCH_COMPLETED)
+            def print_param():
+                if (engine.state.epoch)%5 == 0:
```
Comment (sdesrozis) [inline ignite/handlers/state_param_scheduler.py:357]:
```suggestion
                if engine.state.epoch % 5 == 0:
```

File: ignite/handlers/state_param_scheduler.py [hunk 5]
```diff
@@ -329,21 +329,43 @@ class StepStateScheduler(StateParamScheduler):
...
+                    print(f"epoch = {engine.state.epoch}, param = {engine.state.param}")
```
Comment (sdesrozis) [inline ignite/handlers/state_param_scheduler.py:358]:
```suggestion
                    print(engine.state.param)
```

File: ignite/handlers/state_param_scheduler.py [hunk 6]
```diff
@@ -329,21 +329,43 @@ class StepStateScheduler(StateParamScheduler):
...
 
-            engine.run([0] * 8, max_epochs=10)
+            engine.run([0], max_epochs=25)
+
+        .. testoutput::
+
+            epoch = 5, param = 0.9
```
Comment (sdesrozis) [inline ignite/handlers/state_param_scheduler.py:364]:
```suggestion
            0.9
```

File: ignite/handlers/state_param_scheduler.py [hunk 7]
```diff
@@ -329,21 +329,43 @@ class StepStateScheduler(StateParamScheduler):
...
+            epoch = 10, param = 0.81
```
Comment (sdesrozis) [inline ignite/handlers/state_param_scheduler.py:365]:
```suggestion
            0.81
```

File: ignite/handlers/state_param_scheduler.py [hunk 8]
```diff
@@ -329,21 +329,43 @@ class StepStateScheduler(StateParamScheduler):
...
+            epoch = 15, param = 0.7290...
```
Comment (sdesrozis) [inline ignite/handlers/state_param_scheduler.py:366]:
```suggestion
            0.7290...
```

File: ignite/handlers/state_param_scheduler.py [hunk 9]
```diff
@@ -329,21 +329,43 @@ class StepStateScheduler(StateParamScheduler):
...
+            epoch = 20, param = 0.6561
```
Comment (sdesrozis) [inline ignite/handlers/state_param_scheduler.py:367]:
```suggestion
            0.6561
```

File: ignite/handlers/state_param_scheduler.py [hunk 10]
```diff
@@ -329,21 +329,43 @@ class StepStateScheduler(StateParamScheduler):
...
+            epoch = 25, param = 0.5904...
```
Comment (sdesrozis) [inline ignite/handlers/state_param_scheduler.py:368]:
```suggestion
            0.5904...
```

**Comment by sdesrozis** [comment]
@bibhabasumohapatra Thanks ! I left a few minor comments to address. The point is to be consistent witth the previous PR.

**Comment by bibhabasumohapatra** [comment]
> @bibhabasumohapatra Thanks ! I left a few minor comments to address. The point is to be consistent with the previous PR.

wanted to point out epoch ... for clarity, but yes Consistency is valid point.

File: ignite/handlers/state_param_scheduler.py [hunk 11]
```diff
@@ -329,21 +329,42 @@ class StepStateScheduler(StateParamScheduler):
...
+        .. testcode::
 
-            ...
-            engine = Engine(train_step)
 
             param_scheduler = StepStateScheduler(
-                param_name="param", initial_value=10, gamma=0.99, step_size=5
-            )
+                param_name="param", initial_value=1, gamma=0.9, step_size=5, create_new=True
+                )
```
Comment (sdesrozis) [inline ignite/handlers/state_param_scheduler.py:341]:
```suggestion
            )
```

File: ignite/handlers/state_param_scheduler.py [hunk 12]
```diff
@@ -329,21 +329,42 @@ class StepStateScheduler(StateParamScheduler):
...
+            # Epoch 10, param changes from 0.9 to 0.9*0.9, param = 0.81
+            # Epoch 10 to 14, param = 0.81, as step size is 5
+            # Epoch 15, param changes from 0.81 to 0.81*0.9, param = 0.729
+            # and so on ... the param change at Epoch = 5, 10, 15, 20, . . .
 
             param_scheduler.attach(engine, Events.EPOCH_COMPLETED)
 
-            # basic handler to print scheduled state parameter
-            engine.add_event_handler(Events.EPOCH_COMPLETED, lambda _ : print(engine.state.param))
+            @engine.on(Events.EPOCH_COMPLETED)
```
Comment (sdesrozis) [inline ignite/handlers/state_param_scheduler.py:354]: See the doc https://pytorch-ignite.ai/concepts/02-events-and-handlers/

We can use event filter:
```python
@engine.on(Events.EPOCH_COMPLETED(every=5))
def print_param():
    print(engine.state.param)
```

File: ignite/handlers/state_param_scheduler.py [hunk 13]
```diff
@@ -329,21 +329,42 @@ class StepStateScheduler(StateParamScheduler):
 
     Examples:
 
-        .. code-block:: python
+        .. testsetup::
+
+            engine = get_default_trainer()
```
Comment (sdesrozis) [inline ignite/handlers/state_param_scheduler.py:334]:
```suggestion
            default_trainer = get_default_trainer()
```

Sorry, I realized that `default_trainer` would be the engine name. Could you rename in the other examples in `ExpStateScheduler` too ?

Comment (bibhabasumohapatra) [reply ignite/handlers/state_param_scheduler.py:334]: ok. will do

File: ignite/handlers/state_param_scheduler.py (hunk 12 above)
Comment (sdesrozis) [reply ignite/handlers/state_param_scheduler.py]: @bibhabasumohapatra Could you address this ?

**Review by sdesrozis — CHANGES_REQUESTED** [review]
Please, address the comment https://github.com/pytorch/ignite/pull/2394#discussion_r776216211

File: ignite/handlers/state_param_scheduler.py (hunk 12 above)
Comment (bibhabasumohapatra) [reply ignite/handlers/state_param_scheduler.py]: If i do this... Then  it will 25 lines!... And  only  wanted to print changes!

Comment (bibhabasumohapatra) [reply ignite/handlers/state_param_scheduler.py]: cool!  knew (every=5) but did not thought about it here !!  thanks again.

File: ignite/handlers/state_param_scheduler.py [hunk 14]
```diff
@@ -344,21 +344,41 @@ class StepStateScheduler(StateParamScheduler):
 
     Examples:
 
-        .. code-block:: python
+        .. testsetup::
+
+            default_trainer = get_default_trainer()
+
+        .. testcode::
```
Comment (sdesrozis) [inline ignite/handlers/state_param_scheduler.py:352]:
```suggestion
```

**Review by sdesrozis — APPROVED** [review]
@bibhabasumohapatra Thanks, LGTM !
